tests/variables/test_precipitation.py

```diff
@@ -89,8 +91,80 @@ def test_convert_removes_negative_rates(self):
+        expected_precip_rate = xr.DataArray(expected_precip_rate_values, dims="time", coords={"time": time[1:]})
+
+        result = precipitation.get_rate(precip_accumulated)
+
+        xr.testing.assert_equal(result, expected_precip_rate)
+
+    def test_irregular_sample_rates(self):
+        """
+        There can be occasions where the sample rate is not regular.
+        E.g., on day 300 or after merging multiple datasets.
+
+        # TODO: Should we allow this?
+        # TODO: It is unclear if the rate should be calculated per hour or for the sample window. Should the last value be 7mm or 7/24 mm/h
+        """
+        precip_accumulated_values =   [0.0, 1.0,  1.0,  3.0, 10.0]
+        expected_precip_rate_values = [     1.0,  0.0,  2.0,  7.0]
+        time = pd.to_datetime('2023-10-26') + pd.to_timedelta(['21:00:00', '22:00:00', '23:00:00', '24:00:00', '48:00:00'])
+        expected_precip_rate = xr.DataArray(
+            expected_precip_rate_values, dims="time", coords={"time": time[1:]}
+        )
+        precip_accumulated = xr.DataArray(
+            precip_accumulated_values, dims="time", coords={"time": time}
+        )
+
+        result = precipitation.get_rate(precip_accumulated)
+        xr.testing.assert_equal(result, expected_precip_rate)
+
+    def test_sub_hourly_rates(self):
+        """
+        This is an example where the sample rate higher than h^-1
+        It is unclear if the rate should be calculated per hour or for the sample window
+        """
+        n_samples = 100
+        precip_accumulated_values = np.cumsum(np.random.rand(n_samples))
+        expected_precip_rate_values = np.diff(precip_accumulated_values)
+        time = pd.date_range("2025-06-01", periods=n_samples, freq="600s")
+        precip_accumulated = xr.DataArray(
+            precip_accumulated_values, dims="time", coords={"time": time}
+        )
+        expected_precip_rate = xr.DataArray(
+            expected_precip_rate_values, dims="time", coords={"time": time[1:]}
+        )
+
+        result = precipitation.get_rate(precip_accumulated)
+        xr.testing.assert_equal(result, expected_precip_rate)
+
+    def test_sup_hourly_rates(self):
+        """
+        This is an example where the sample is daily - lower than h^-1
+        It is unclear if the rate should be calculated per hour or for the sample window
+        """
+        precip_accumulated_values = [0.0, 4.0, 4.0, 5.0, 6.0]
+        expected_precip_rate_values = [3.0, 0.0, 1.0, 1.0]
+        time = pd.date_range("2025-06-01", periods=len(precip_accumulated_values), freq="1d")
+        expected_precip_rate = xr.DataArray(
+            expected_precip_rate_values, dims="time", coords={"time": time[1:]}
+        )
+        precip_accumulated = xr.DataArray(
+            precip_accumulated_values, dims="time", coords={"time": time}
+        )
+
+        result = precipitation.get_rate(precip_accumulated)
+        xr.testing.assert_equal(result, expected_precip_rate)
```

Comment:
This test does not correspond to an existing setting: There is no station with Lufft that transmit daily values. Still the code should perform well if they occured. Just wondering if it needs to be tested.

Reply:
I have removed this test.